pkg/vm/precompiles.go

```diff
@@ -167,67 +170,70 @@ func getPrecompiles(config params.ChainConfig) (params.ChainConfig, error) {
-			return config, err
+			return config, errored, err
+		}
+
+		switch addPrecompile {
+		case prompts.No:
+			return config, stageAfterPrecompile, nil
+		case goBackMsg:
+			return config, stageBeforePrecompile, nil
 		}
 
-		if addPrecompile {
-			precompileDecision, err := prompts.CaptureList(
-				"Choose precompile",
-				remainingPrecompiles,
-			)
+		precompileDecision, err := prompts.CaptureList(
+			"Choose precompile",
+			remainingPrecompiles,
+		)
+		if err != nil {
+			return config, errored, err
+		}
+
+		switch precompileDecision {
+		case nativeMint:
+			mintConfig, cancelled, err := configureMinterList()
 			if err != nil {
-				return config, err
+				return config, errored, err
 			}
-
-			switch precompileDecision {
-			case nativeMint:
-				mintConfig, cancelled, err := configureMinterList()
+			if !cancelled {
+				config.ContractNativeMinterConfig = mintConfig
+				remainingPrecompiles, err = removePrecompile(remainingPrecompiles, nativeMint)
 				if err != nil {
-					return config, err
-				}
-				if !cancelled {
-					config.ContractNativeMinterConfig = mintConfig
-					remainingPrecompiles, err = removePrecompile(remainingPrecompiles, nativeMint)
-					if err != nil {
-						return config, err
-					}
+					return config, errored, err
 				}
-			case contractAllowList:
-				contractConfig, cancelled, err := configureContractAllowList()
+			}
+		case contractAllowList:
+			contractConfig, cancelled, err := configureContractAllowList()
+			if err != nil {
+				return config, errored, err
+			}
+			if !cancelled {
+				config.ContractDeployerAllowListConfig = contractConfig
+				remainingPrecompiles, err = removePrecompile(remainingPrecompiles, contractAllowList)
 				if err != nil {
-					return config, err
-				}
-				if !cancelled {
-					config.ContractDeployerAllowListConfig = contractConfig
-					remainingPrecompiles, err = removePrecompile(remainingPrecompiles, contractAllowList)
-					if err != nil {
-						return config, err
-					}
+					return config, errored, err
 				}
-			case txAllowList:
-				txConfig, cancelled, err := configureTransactionAllowList()
+			}
+		case txAllowList:
+			txConfig, cancelled, err := configureTransactionAllowList()
+			if err != nil {
+				return config, errored, err
+			}
+			if !cancelled {
+				config.TxAllowListConfig = txConfig
+				remainingPrecompiles, err = removePrecompile(remainingPrecompiles, txAllowList)
 				if err != nil {
-					return config, err
+					return config, errored, err
 				}
-				if !cancelled {
-					config.TxAllowListConfig = txConfig
-					remainingPrecompiles, err = removePrecompile(remainingPrecompiles, txAllowList)
-					if err != nil {
-						return config, err
-					}
-				}
-			case cancel:
-				return config, nil
-			}
-
-			if len(remainingPrecompiles) == 1 {
-				return config, nil
 			}
+		case cancel:
+			return config, stageAfterPrecompile, nil
+		}
 
-		} else {
-			return config, nil
+		if len(remainingPrecompiles) == 1 {
```

Comment:
I added a comment, but that's the state when all precompiles have been configured. There's a `cancel` option in the list, so that's the one element that's still in there.